---
title: "CRC_KRAS_Age"
author: "Shruti Kakan"
date: "2024-05-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("TCGAbiolinks")
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("limma")
library("limma")
library("edgeR")
library("glmnet")
library("factoextra")
library("FactoMineR")
library("caret")
library("SummarizedExperiment")
library("gplots")
library("survival")
library("survminer")
library("RColorBrewer")
library("gProfileR")
library("genefilter")

setwd("~/Documents/TCGA/TCGA/")
```

## R Markdown

Number of cases by Genomics Data Type for Colon adenocarcinoma
```{r}
#TCGA Data
library(TCGAbiolinks)
GDCprojects = getGDCprojects()
head(GDCprojects[c("project_id", "name")])
TCGAbiolinks:::getProjectSummary("TCGA-COAD")

#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("TCGAbiolinks")

```

Number of cases by Genomics Data Type for Rectal Adenocarcinoma
```{r}
TCGAbiolinks:::getProjectSummary("TCGA-READ")
```

### Obtaining RNA-Sequencing and clinical data for patients

```{r}
CRC_RNAseq = GDCquery(
  project = c("TCGA-COAD", "TCGA-READ" ),
  data.category = "Transcriptome Profiling", 
  data.type = "Gene Expression Quantification",
  experimental.strategy = "RNA-Seq",
  workflow.type = "STAR - Counts")

crc_res = getResults(CRC_RNAseq) 
colnames(crc_res)
# first 6 types of tumor tissue
head(crc_res$sample_type)

# summary of distinct tissues types present in this study
summary(factor(crc_res$sample_type))
```
There are 51 normal tissue samples, 647 primary CRC tumor types, 1 Metastatic and 2 Recurrent Tumor types. Filtering the dataset to include only Primary and Normal solid tissue types.

```{r}
CRC_TCGA = GDCquery(
  project = c("TCGA-COAD", "TCGA-READ"),
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  experimental.strategy = "RNA-Seq",
  workflow.type = "STAR - Counts",
  sample.type = c("Primary Tumor", "Solid Tissue Normal"))

#GDCdownload(query = CRC_TCGA)
crc_data = GDCprepare(CRC_TCGA)
```
```{r}
dim(crc_data)
colnames(colData(crc_data))
(crc_data$age_at_diagnosis) #Age may be in number of days
plot(crc_data$age_at_diagnosis/365, crc_data$age_at_index) 

```


We have data from 698 patients. Both clinical and gene expression data are present in this object.
There is also close correlation between the age of diagnosis and age at which samples were indexed.

```{r}
table(crc_data@colData$vital_status)
```
Of these 549 patients are still alive. 

```{r}
table(crc_data@colData$ajcc_pathologic_stage)
crc_data$prior_treatment
crc_data$treatments[[5]]$treatment_type

```

Tumor Staging Data is present for the CRC tumors.
To an extent, treatment information is also present

```{r}
table(crc_data@colData$tissue_or_organ_of_origin)
```
Tissue of origin tells us what type of cancer the patient has.

```{r}
table(crc_data@colData$gender)
```


```{r}
#We have clinical information of the patients
table(crc_data@colData$age_at_diagnosis/365)
table(crc_data@colData$age_at_index)
histogram((crc_data@colData$age_at_index), 
          xlab = "Age",
          breaks=20)
```


```{r}

```


### Obtain Colon Adenocarconoma SNV datasets COAD
Querying all Single Nucleotide Variation data from COAD project
READ is rectal adenocarcinoma
```{r}
query_COAD = GDCquery(
  project = "TCGA-COAD",
  data.category = "Simple Nucleotide Variation",
  data.type = "Masked Somatic Mutation" )
query_READ = GDCquery(
  project =  "TCGA-READ",
  data.category = "Simple Nucleotide Variation",
  data.type = "Masked Somatic Mutation")

View(getResults(query_COAD))
View(getResults(query_READ))
```

### Querying Data for Somatic Mutations in Colon and Rectum Adenocarcinoma
```{r}
coad_res = getResults(query_COAD) # make results as table
# head(lihc_res) # data of the first 6 patients.
colnames(coad_res) # columns present in the table
head(coad_res$sample_type) # first 6 types of tissue.
summary(factor(coad_res$sample_type)) # summary of distinct tissues types present in this study
summary(factor(coad_res$))

read_res = getResults(query_READ) # make results as table
# head(lihc_res) # data of the first 6 patients.
colnames(read_res) # columns present in the table
head(read_res$sample_type) # first 6 types of tissue.
summary(factor(read_res$sample_type)) # summary of distinct tissues types present in this study

```

There are 82 normal tissue samples

Downloading the somatic mutations Data File data file
```{r}
GDCdownload(query = query_COAD)
GDCdownload(query = query_READ)
```

#Loading the data in R
```{r}
coad_data = GDCprepare(query_COAD)
read_data = GDCprepare(query_READ)

install.packages("remotes")
remotes::install_github("nlhuong/TimeSeriesExperiment")

library("TimeSeriesExperiment")
dim(coad_data)
dim(read_data)
library(SummarizedExperiment)
#colnames(SummarizedExperiment::colData(coad_data))
#colnames(colData(read_data))

```


```{r}

```


```{r}

```


```{r}

```

