---
title: "CRC_KRAS_Age"
author: "Shruti Kakan"
date: "2024-05-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("TCGAbiolinks")
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("limma")
library("limma")
library("edgeR")
library("glmnet")
library("factoextra")
library("FactoMineR")
library("caret")
library("SummarizedExperiment")
library("gplots")
library("survival")
library("survminer")
library("RColorBrewer")
library("gProfileR")
#library("genefilter")

setwd("~/Documents/TCGA/TCGA/")
```

## R Markdown

Number of cases by Genomics Data Type for Colon adenocarcinoma
```{r}
#TCGA Data
library(TCGAbiolinks)
GDCprojects = getGDCprojects()
head(GDCprojects[c("project_id", "name")])
TCGAbiolinks:::getProjectSummary("TCGA-COAD")

#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("TCGAbiolinks")

```

Number of cases by Genomics Data Type for Rectal Adenocarcinoma
```{r}
TCGAbiolinks:::getProjectSummary("TCGA-READ")
```

### Obtaining RNA-Sequencing and clinical data for patients

```{r}
CRC_RNAseq = GDCquery(
  project = c("TCGA-COAD", "TCGA-READ" ),
  data.category = "Transcriptome Profiling", 
  data.type = "Gene Expression Quantification",
  experimental.strategy = "RNA-Seq",
  workflow.type = "STAR - Counts")

crc_res = getResults(CRC_RNAseq) 
colnames(crc_res)
# first 6 types of tumor tissue
head(crc_res$sample_type)

# summary of distinct tissues types present in this study
summary(factor(crc_res$sample_type))
```

There are 51 normal tissue samples, 647 primary CRC tumor types, 1 Metastatic and 2 Recurrent Tumor types. Filtering the dataset to include only Primary and Normal solid tissue types.

```{r}
CRC_TCGA = GDCquery(
  project = c("TCGA-COAD", "TCGA-READ"),
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  experimental.strategy = "RNA-Seq",
  workflow.type = "STAR - Counts",
  sample.type = c("Primary Tumor", "Solid Tissue Normal"))

#GDCdownload(query = CRC_TCGA)
crc_data = GDCprepare(CRC_TCGA)
```
```{r}
dim(crc_data)
colnames(colData(crc_data))
(crc_data$age_at_diagnosis) #Age may be in number of days
plot(crc_data$age_at_diagnosis/365, crc_data$age_at_index) 

```

We have data from 698 patients. Both clinical and gene expression data are present in this object.
There is also close correlation between the age of diagnosis and age at which samples were indexed.

```{r}
table(crc_data@colData$vital_status)
```
Of these 549 patients are still alive. 

```{r}
table(crc_data@colData$ajcc_pathologic_stage)
crc_data$prior_treatment
crc_data$treatments[[5]]$treatment_type

```

Tumor Staging Data is present for the CRC tumors.
To an extent, treatment information is also present

```{r}
table(crc_data@colData$tissue_or_organ_of_origin)
crc_data@colData
```
Tissue of origin tells us what type of cancer the patient has.

```{r}
table(crc_data@colData$gender)
```


```{r}
#We have clinical information of the patients
table(crc_data@colData$age_at_diagnosis/365)
table(crc_data@colData$age_at_index)
histogram((crc_data@colData$age_at_index), 
          xlab = "Age",
          breaks=20)
```
Most CRC patients fall in the age range of 65-75 years

```{r}
table(crc_data@colData$race)
```
We have race information for nearly half of the patients.

```{r}
dim(assay(crc_data))
head(assay(crc_data)[,1:10]) # expression of first 6 genes and first 10 samples

head(rowData(crc_data))#ensembl id and gene id of the first 6 genes.

(rowData(crc_data))
#head(dge$counts[,which(dge$genes[,"gene_name"])])

# Save the data as a file, if you need it later, you can just load this file
# instead of having to run the whole pipeline again
saveRDS(object = crc_data,
        file = "crc_RNAseq_data.RDS",
        compress = FALSE)

crc_data <- readRDS("crc_RNAseq_data.RDS", refhook = NULL)
```


```{r Limma Pipeline for identifying DE Genes}
clinical_data = colData(crc_data)
group = as.factor(clinical_data$definition)
group = relevel(group, ref="Solid Tissue Normal")

design = model.matrix(~group)
head(design)

#Remove genes, which have low amount of counts
dge = DGEList( # creating a DGEList object
  counts=assay(crc_data),
  samples=colData(crc_data),
  group=factor(crc_data@colData$sample_type),
  genes=as.data.frame(rowData(crc_data)))
library(DEFormats)
dds = as.DESeqDataSet(dge)

# filtering
keep = filterByExpr(dge, design, min.count=10) # defining which genes to keep
length(which(keep))
length(which(keep==FALSE))
dge = dge[keep,,keep.lib.sizes=FALSE] # filtering the dge object
length(which(keep))
length(which(keep==FALSE))
rm(keep) #  use rm() to remove objects from memory if you don't need them anymore

dge = calcNormFactors(dge,method="TMM")
v = voom(dge,design,plot=TRUE)
#using lmFit lets fit a series of linear models, one to each of the probes. These data will then be fed to eBayes to produce a complex object which holds a number of statistics that we can use to rank the differentially expressed genes.
fit = lmFit(v, design)
fit = eBayes(fit)

topGenes = topTable(fit, coef=1, sort.by="p", number = 7000)
print(topGenes)

```


```{r}

```


### Obtain Colon Adenocarconoma SNV datasets COAD
Querying all Single Nucleotide Variation data from COAD project
READ is rectal adenocarcinoma
```{r}
CRC_Mut = GDCquery(
  project = c("TCGA-COAD","TCGA-READ"),
  data.category = "Simple Nucleotide Variation",
  data.type = "Masked Somatic Mutation")

View(getResults(CRC_Mut))

# Get all patients that have DNA methylation and gene expression.
common.patients <- intersect(
    substr(getResults(CRC_TCGA, cols = "cases"), 1, 12),
    substr(getResults(CRC_Mut, cols = "cases"), 1, 12))
head(common.patients)

```

### Querying Data for Somatic Mutations in Colon and Rectum Adenocarcinoma
```{r}
CRC_Mut_res = getResults(CRC_Mut) # make results as table
colnames(CRC_Mut) # columns present in the table
head(CRC_Mut$sample_type) # first 6 types of tissue.
summary(factor(CRC_Mut_res$sample_type)) # summary of distinct tissues types present in this study

```

There are 82 normal tissue samples
`
Downloading the somatic mutations Data File
```{r}
GDCdownload(query = CRC_Mut)

crc_mut <- GDCprepare(CRC_Mut)
```

#Loading the data in R
```{r}
CRC_Mut_data = GDCprepare(CRC_Mut)
dim(CRC_Mut_data)
```


```{r}
which(CRC_Mut_data$HGVSp_Short=="p.G12D")

crc_mut_sub1 <- CRC_Mut_data[unlist(lapply(common.patients, function(x) grep(x, CRC_Mut_data$Tumor_Sample_Barcode))),]

crc_mut_sub2 <- crc_mut_sub1[which(crc_mut_sub1$Hugo_Symbol=="KRAS"),]
crc_mut_sub3 <- crc_mut_sub2[which(crc_mut_sub2$HGVSp_Short=="p.G12D"),]

crc_mut_sub1$patient <- substr(crc_mut_sub1$Tumor_Sample_Barcode, 1, 12)
crc_mut_sub2$patient <- substr(crc_mut_sub2$Tumor_Sample_Barcode, 1, 12)
crc_mut_sub3$patient <- substr(crc_mut_sub3$Tumor_Sample_Barcode, 1, 12)

head(crc_data@colData$patient)
head(crc_data@colData$age_at_index)
head(crc_data@colData$age_at_diagnosis)
head(crc_data@colData$gender)
head(crc_data@colData$ajcc_pathologic_stage)
head(crc_data@colData$race)
head(crc_data@colData$tissue_or_organ_of_origin)

pt_id_age <- as.data.frame(cbind(crc_data@colData$patient, crc_data@colData$age_at_index, crc_data@colData$age_at_diagnosis, crc_data@colData$gender, crc_data@colData$ajcc_pathologic_stage, crc_data@colData$race,crc_data@colData$tissue_or_organ_of_origin))
colnames(pt_id_age) <- c("patient", "age_at_index", "age_at_diagnosis", "gender", "Tumor_Stage", "Race", "Tumor_Location")
pt_id_age$age_at_index <- as.numeric(pt_id_age$age_at_index)
pt_id_age$age_at_diagnosis <- as.numeric(pt_id_age$age_at_diagnosis)
pt_id_age$age_at_diagnosis <- (pt_id_age$age_at_diagnosis)/365
pt_id_age$gender <- as.factor(pt_id_age$gender)
pt_id_age$Tumor_Stage <- as.factor(pt_id_age$Tumor_Stage)
pt_id_age$Race <- as.factor(pt_id_age$Race)
pt_id_age$Tumor_Location <- as.factor(pt_id_age$Tumor_Location)


library(dplyr)
crc_mut_sub1 <- full_join(crc_mut_sub1, pt_id_age, by="patient", relationship = "many-to-many")

crc_mut_sub2 <- full_join(crc_mut_sub2, pt_id_age, by="patient", relationship = "many-to-many")
crc_mut_sub2[which(is.na(crc_mut_sub2$Hugo_Symbol)),"Hugo_Symbol"] <- "No_KRAS"

crc_mut_sub3 <- full_join(crc_mut_sub3, pt_id_age, by="patient", relationship = "many-to-many")
crc_mut_sub3[which(is.na(crc_mut_sub3$Hugo_Symbol)),"Hugo_Symbol"] <- "No_KRAS"
crc_mut_sub3$Hugo_Symbol <- as.factor(crc_mut_sub3$Hugo_Symbol)
crc_mut_sub3$patient <- as.factor(crc_mut_sub3$patient)
crc_mut_sub3$HGVSp_Short <- as.factor(crc_mut_sub3$HGVSp_Short)
library(ggsci)
```

#### CRC patient sex vs age
```{r}
ggplot(crc_mut_sub3, aes(x=age_at_diagnosis, color=Hugo_Symbol)) +
  geom_histogram(binwidth = 5, fill="white") +
  labs(title="Distribution of CRC patients with KRAS mutations") +
  scale_color_jco() +
  theme_minimal()

ggplot(crc_mut_sub3, aes(x=age_at_diagnosis, fill=HGVSp_Short, color=Hugo_Symbol)) +
  geom_histogram(binwidth = 10) +
  scale_fill_aaas() +
  scale_color_jco() +
  theme_minimal()
```


```{r}
colourCount = length(unique(crc_mut_sub2$HGVSp_Short))
getPalette = colorRampPalette(brewer.pal(9, "Spectral"))
ggplot(crc_mut_sub2, aes(x=age_at_diagnosis, color=Hugo_Symbol, fill=HGVSp_Short)) +
  geom_histogram(binwidth = 5) +
  xlab("Age at CRC Diagnosis") + ylab("Number of Patients") +
  scale_fill_manual(values = colorRampPalette(brewer.pal(12, "Accent"))(colourCount), na.value = "white") +
  #labs(title="Distribution of CRC patients") +
  scale_color_jco(na.value = "white") +
  theme_minimal()
```

### Patient distribution with Tumor Location
```{r}
colourCount = length(unique(crc_mut_sub2$HGVSp_Short))
getPalette = colorRampPalette(brewer.pal(9, "Spectral"))
ggplot(crc_mut_sub2, aes(x=Tumor_Location, color=Hugo_Symbol)) +
  geom_bar(aes(fill=HGVSp_Short), position="fill") +
  xlab("Tumor Location") + ylab("Number of Patients") +
  scale_fill_manual(values = colorRampPalette(brewer.pal(12, "Accent"))(colourCount), na.value = "white") +
  scale_color_jco(na.value = "white") +
  theme_minimal() +
  scale_y_continuous(labels=scales::percent) +
   scale_x_discrete(guide = guide_axis(n.dodge = 4))
```


### Patient distribution with Race
```{r}
ggplot(crc_mut_sub2, aes(x=Race, color=Hugo_Symbol, fill=HGVSp_Short)) +
  geom_bar() +
  xlab("Patient Race") + ylab("Number of Patients") +
  scale_fill_manual(values = colorRampPalette(brewer.pal(12, "Accent"))(colourCount), na.value = "white") +
  scale_color_jco(na.value = "white") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```


#### Code for KRAS mutations with Tumor staging
```{r}
colourCount = length(unique(crc_mut_sub2$HGVSp_Short))
getPalette = colorRampPalette(brewer.pal(9, "Spectral"))
ggplot(crc_mut_sub2, aes(x=Tumor_Stage, color=Hugo_Symbol, fill=HGVSp_Short)) +
  geom_bar() +
  xlab(" CRC Stage") + ylab("Number of Patients") +
  scale_fill_manual(values = colorRampPalette(brewer.pal(12, "Accent"))(colourCount), na.value = "white") +
  #labs(title="Distribution of CRC patients") +
  scale_color_jco(na.value = "white") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```



#### Code for BRAF and it's variants
```{r}
#which(CRC_Mut_data$HGVSp_Short=="p.G12D")

crc_mut_sub1 <- CRC_Mut_data[unlist(lapply(common.patients, function(x) grep(x, CRC_Mut_data$Tumor_Sample_Barcode))),]

crc_mut_braf2 <- crc_mut_sub1[which(crc_mut_sub1$Hugo_Symbol=="BRAF"),]

crc_mut_sub1$patient <- substr(crc_mut_sub1$Tumor_Sample_Barcode, 1, 12)
crc_mut_braf2$patient <- substr(crc_mut_braf2$Tumor_Sample_Barcode, 1, 12)

pt_id_age <- as.data.frame(cbind(crc_data@colData$patient, crc_data@colData$age_at_index, crc_data@colData$age_at_diagnosis, crc_data@colData$gender, crc_data@colData$ajcc_pathologic_stage, crc_data@colData$race))
colnames(pt_id_age) <- c("patient", "age_at_index", "age_at_diagnosis", "gender", "Tumor_Stage", "Race")
pt_id_age$age_at_index <- as.numeric(pt_id_age$age_at_index)
pt_id_age$age_at_diagnosis <- as.numeric(pt_id_age$age_at_diagnosis)
pt_id_age$age_at_diagnosis <- (pt_id_age$age_at_diagnosis)/365
pt_id_age$gender <- as.factor(pt_id_age$gender)
pt_id_age$Tumor_Stage <- as.factor(pt_id_age$Tumor_Stage)
pt_id_age$Race <- as.factor(pt_id_age$Race)


library(dplyr)
crc_mut_sub1 <- full_join(crc_mut_sub1, pt_id_age, by="patient", relationship = "many-to-many")

crc_mut_braf2 <- full_join(crc_mut_braf2, pt_id_age, by="patient", relationship = "many-to-many")
crc_mut_braf2[which(is.na(crc_mut_braf2$Hugo_Symbol)),"Hugo_Symbol"] <- "No_BRAF"

crc_mut_braf2$Hugo_Symbol <- as.factor(crc_mut_braf2$Hugo_Symbol)
crc_mut_braf2$patient <- as.factor(crc_mut_braf2$patient)
crc_mut_braf2$HGVSp_Short <- as.factor(crc_mut_braf2$HGVSp_Short)
library(ggsci)
```

```{r}
colourCount = length(unique(crc_mut_braf2$HGVSp_Short))
getPalette = colorRampPalette(brewer.pal(9, "Spectral"))
ggplot(crc_mut_braf2, aes(x=age_at_diagnosis, color=Hugo_Symbol, fill=HGVSp_Short)) +
  geom_histogram(binwidth = 10) +
  xlab("Age at CRC Diagnosis") + ylab("Number of Patients") +
  scale_fill_manual(values = colorRampPalette(brewer.pal(12, "Accent"))(colourCount), na.value = "white") +
  #labs(title="Distribution of CRC patients") +
  scale_color_jco(na.value = "white") +
  theme_minimal()
```



```{r}

```

#### Code for VEGFA mutations with Tumor staging
```{r}

crc_mut_sub1 <- CRC_Mut_data[unlist(lapply(common.patients, function(x) grep(x, CRC_Mut_data$Tumor_Sample_Barcode))),]
crc_mut_vegf2 <- crc_mut_sub1[which(crc_mut_sub1$Hugo_Symbol=="VEGFA"),]
crc_mut_sub1$patient <- substr(crc_mut_sub1$Tumor_Sample_Barcode, 1, 12)
crc_mut_vegf2$patient <- substr(crc_mut_vegf2$Tumor_Sample_Barcode, 1, 12)
pt_id_age <- as.data.frame(cbind(crc_data@colData$patient, crc_data@colData$age_at_index, crc_data@colData$age_at_diagnosis, crc_data@colData$gender, crc_data@colData$ajcc_pathologic_stage, crc_data@colData$race))
colnames(pt_id_age) <- c("patient", "age_at_index", "age_at_diagnosis", "gender", "Tumor_Stage", "Race")
pt_id_age$age_at_index <- as.numeric(pt_id_age$age_at_index)
pt_id_age$age_at_diagnosis <- as.numeric(pt_id_age$age_at_diagnosis)
pt_id_age$age_at_diagnosis <- (pt_id_age$age_at_diagnosis)/365
pt_id_age$gender <- as.factor(pt_id_age$gender)
pt_id_age$Tumor_Stage <- as.factor(pt_id_age$Tumor_Stage)
pt_id_age$Race <- as.factor(pt_id_age$Race)


library(dplyr)
crc_mut_sub1 <- full_join(crc_mut_sub1, pt_id_age, by="patient", relationship = "many-to-many")

crc_mut_vegf2 <- right_join(crc_mut_vegf2, pt_id_age, by="patient", relationship = "many-to-many")
crc_mut_vegf2[which(is.na(crc_mut_vegf2$Hugo_Symbol)),"Hugo_Symbol"] <- "No_VEGF"

crc_mut_vegf2$Hugo_Symbol <- as.factor(crc_mut_vegf2$Hugo_Symbol)
crc_mut_vegf2$patient <- as.factor(crc_mut_vegf2$patient)
crc_mut_vegf2$HGVSp_Short <- as.factor(crc_mut_vegf2$HGVSp_Short)
library(ggsci)

colourCount = length(unique(crc_mut_vegf2$HGVSp_Short))
getPalette = colorRampPalette(brewer.pal(9, "Spectral"))
ggplot(crc_mut_sub1, aes(x=Tumor_Stage, color=Hugo_Symbol, fill=HGVSp_Short)) +
  geom_bar() +
  xlab(" CRC Stage") + ylab("Number of Patients") +
  #scale_fill_manual(values = colorRampPalette(brewer.pal(12, "Accent"))(colourCount), na.value = "white") +
  #labs(title="Distribution of CRC patients") +
  scale_color_jco(na.value = "white") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```


##### Gene Expression Data on KDR (VEGFR1 Receptor)
```{r}
expr_diseased = d_mat[rownames(clin_df), gene_id]
expr_healthy = d_mat[setdiff(rownames(d_mat), rownames(clin_df)), gene_id]

boxplot(expr_diseased, expr_healthy,
        names=c("Diseased", "Healthy"), main="Distribution of gene expression")

rowData(crc_data[which(crc_data@rowRanges$gene_name=="KDR")])
KDR_clinical <- crc_data@colData[,c("sample_type", "patient", "barcode", "ajcc_pathologic_stage")]
KDR_clinical <- as.data.frame(KDR_clinical@listData, row.names = KDR_clinical@rownames )
KDR_clinical$Sample_ID <- rownames(KDR_clinical)
KDR <- as.data.frame(assay(crc_data)["ENSG00000128052.10",])
KDR$Sample_ID <- rownames(KDR)
KDR_dataset = full_join(KDR, KDR_clinical, by = "Sample_ID")
str(KDR_dataset)
KDR_dataset$patient <- as.factor(KDR_dataset$patient)
KDR_dataset$sample_type <- as.factor(KDR_dataset$sample_type)
KDR_dataset$barcode <- as.factor(KDR_dataset$barcode)
KDR_dataset$ajcc_pathologic_stage <- as.factor(KDR_dataset$ajcc_pathologic_stage)

ggplot(KDR_dataset, aes(x=ajcc_pathologic_stage, y=log10(`assay(crc_data)["ENSG00000128052.10", ]`))) +
  geom_boxplot(aes(fill=sample_type)) +
  geom_jitter(alpha=0.5, size=1, width= 0.25, aes(colour = sample_type)) +
  guides(fill="none") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle=90))
```

#VEGFA expression by tumor stage
```{r}
rowData(crc_data[which(crc_data@rowRanges$gene_name=="VEGFA")])
VEGFA_clinical <- crc_data@colData[,c("sample_type", "patient", "barcode", "ajcc_pathologic_stage")]
VEGFA_clinical <- as.data.frame(VEGFA_clinical@listData, row.names = VEGFA_clinical@rownames )
VEGFA_clinical$Sample_ID <- rownames(VEGFA_clinical)
VEGFA <- as.data.frame(assay(crc_data)["ENSG00000112715.25",])
VEGFA$Sample_ID <- rownames(VEGFA)
VEGFA_dataset = full_join(VEGFA, VEGFA_clinical, by = "Sample_ID")
str(VEGFA_dataset)
VEGFA_dataset$patient <- as.factor(VEGFA_dataset$patient)
VEGFA_dataset$sample_type <- as.factor(VEGFA_dataset$sample_type)
VEGFA_dataset$barcode <- as.factor(VEGFA_dataset$barcode)
VEGFA_dataset$ajcc_pathologic_stage <- as.factor(VEGFA_dataset$ajcc_pathologic_stage)

ggplot(VEGFA_dataset, aes(x=sample_type, y=log10(`assay(crc_data)["ENSG00000112715.25", ]`))) +
  geom_boxplot(aes(fill=sample_type)) +
  geom_jitter(alpha=0.5, size=1, width= 0.25, aes(colour = sample_type)) +
  guides(fill="none") +
  theme_minimal() + ylab("Normalized expression of VEGFA") +
  theme(axis.text.x = element_text(angle=90))

ggplot(VEGFA_dataset, aes(x=ajcc_pathologic_stage, y=log10(`assay(crc_data)["ENSG00000112715.25", ]`))) +
  geom_boxplot(aes(fill=sample_type)) +
  geom_jitter(alpha=0.5, size=1, width= 0.25, aes(colour = sample_type)) +
  guides(fill="none") +
  theme_minimal() + ylab("Normalized expression of VEGFA") +
  theme(axis.text.x = element_text(angle=90))
```

Gene Expression by Tumor Stage
```{r}

gene_id <- c("ENSG00000128052.10" , "ENSG00000112715.25", "ENSG00000261371.6", "ENSG00000119888.11")
gene_name <- c("KDR", "VEGFA", "PECAM1", "EPCAM")
Exp_dataset <- list()
Var_X <- c("ajcc_pathologic_stage", "sample_type")

library(dplyr)
for (i in gene_id) {
  print(i)
  Gene_clinical <- crc_data@colData[,c("sample_type", "patient", "barcode", "ajcc_pathologic_stage", "gender", "tissue_or_organ_of_origin")]
  Gene_clinical <- as.data.frame(Gene_clinical@listData, row.names = Gene_clinical@rownames )
  Gene_clinical$Sample_ID <- rownames(Gene_clinical)
  Gene <- as.data.frame(assay(crc_data)[i,])
  Gene$Sample_ID <- rownames(Gene)
  Exp_dataset[[i]] = full_join(Gene, Gene_clinical, by = "Sample_ID")
  Exp_dataset[[i]]$patient <- as.factor(Exp_dataset[[i]]$patient)
  Exp_dataset[[i]]$sample_type <- as.factor(Exp_dataset[[i]]$sample_type)
  Exp_dataset[[i]]$barcode <- as.factor(Exp_dataset[[i]]$barcode)
  Exp_dataset$ajcc_pathologic_stage <- as.factor(Exp_dataset$ajcc_pathologic_stage)
  
  p <- ggplot(Exp_dataset[[i]], aes(x=ajcc_pathologic_stage, y=log10(`assay(crc_data)[i, ]`))) +
  geom_boxplot(aes(fill=sample_type)) +
  geom_jitter(alpha=0.5, size=1, width= 0.25, aes(colour = sample_type, pch=sample_type)) +
  guides(fill="none") +
  theme_minimal() + ylab(paste0("Normalized expression of ", gene_name[which(gene_id==i)])) +
  theme(axis.text.x = element_text(angle=90))
  print(p)
  
  q <- ggplot(Exp_dataset[[i]], aes(x=tissue_or_organ_of_origin, y=log10(`assay(crc_data)[i, ]`))) +
  geom_boxplot(aes(fill=sample_type)) +
  geom_jitter(alpha=0.5, size=1, width= 0.25, aes(colour = factor(tissue_or_organ_of_origin), pch=sample_type)) +
  guides(fill="none") + guides(color="none") +
  theme_minimal() + ylab(paste0("Normalized expression of ", gene_name[which(gene_id==i)])) +
  theme(axis.text.x = element_text(angle=90))
  print(q)
}

```

